handlers/geolocation/geolocation.go

```diff
@@ -26,29 +27,51 @@ type GeolocationHandlersCollection struct {
 
 // redirect an incoming user to a node for playback or 404 handling
 func (c *GeolocationHandlersCollection) RedirectHandler() httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-		nodeHost := c.Config.NodeHost
-		redirectPrefixes := c.Config.RedirectPrefixes
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		host := r.Host
+		prefix, playbackID, pathTmpl, isValid := parsePlaybackID(r.URL.Path)
+
+		if c.Config.CdnRedirectPrefix != "" {
+
+			if !isValid {
+				glog.Warningf("Can not parse playbackID from path %s", r.URL.Path)
+				w.WriteHeader(http.StatusNotFound)
+				return
+			}
 
-		if nodeHost != "" {
-			host := r.Host
-			if host != nodeHost {
-				newURL, err := url.Parse(r.URL.String())
-				if err != nil {
-					glog.Errorf("failed to parse incoming url for redirect url=%s err=%s", r.URL.String(), err)
-					w.WriteHeader(http.StatusInternalServerError)
-					return
-				}
+			doCdnRedirect := contains(playbackID, c.Config.CdnRedirectPlaybackIDs)
+			if doCdnRedirect { // playbackID found in CdnRedirectPlaybackIDs
```

Comment:
super nit: I think this comment does not explain a anything, it basically says the same what the code in the line above.
```suggestion
			if doCdnRedirect {
```